tests/integrationv2/README.md

```diff
@@ -157,3 +157,47 @@ An example of how to test that the server and the client can send and receive ap
 An error similar to this is caused by a runtime error in a test. In `tox.ini` change `-n8` to `-n0` to
 see the actual error causing the OSError.
 
+
+# Criterion
+
+### Why
+
+We wanted to use the rust criterion project to benchmark s2n-tls, without re-writing all the integration tests.
+To accomplish this, we created some criterion benchmarks that use s2nc and s2nd, and a new provider, CriterionS2N, in the python testing framework.
+
+### Prerequisites
+
+Normally, you'd run criterion with `cargo bench --bench <name>`, but cargo does some checks to see if it needs to rebuild
+the benchmark and other housekeeping that slows things down a bit.  Running `cargo bench --no-run` is the benchmark equivalent to `cargo build` and will produce a binary executable.
+
+The CI will run `make install` and `make -C ./bindings/rust` to create and install the s2nc/d binaries in a system-wide location, then build the cargo criterion binary handlers for s2nc and s2nd.
+
+### What is happening inside the CriterionS2N provider?
+
```

Comment:
I don't think this belongs in the README. These kind of implementation details should be commented in the code.

Reply:
moved into the code as a commend.